qa/rpc-tests/signed_blockchain.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2016 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#
+# Test RPC calls related to blockchain state. Tests correspond to code in
+# rpc/blockchain.cpp.
+#
+
+from decimal import Decimal
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.authproxy import JSONRPCException
+from test_framework.util import (
+    assert_equal,
+    start_nodes,
+    assert_raises_jsonrpc,
+)
+
+
+class SignedBlockchainTest(BitcoinTestFramework):
+    """
+    Test signed-blockchain-related RPC calls:
+
+        - getnewblockhex
+        - signblock
+        - combineblocksigs
+        - submitblock
+
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 3
+
+    def setup_network(self, split=False):
+        # Keys for signing 2-of-3
+        pubkeys = ["039560e48d4336e40db447fc136ce24ae1dfdefa5701e4d4e57aa1a1a9f47f3faa", "025a66517c1d85adcd909f9f675bf656708edc4da1f614693e347be6baf0fef4ae", "02ca238faeb3b01d26ae8a39869220dbd84cc8516398afa8958fe613fe4fdf1c04"]
```

Comment:
Perhaps we prefer to calculate keys and pubkeys dynamically like in https://github.com/stevenroose/elements/commit/51a8ee1cdc5ab0217cdb4ff9d44f5bce07131b29 ?
ping @stevenroose

Reply:
Are these the pubkeys associated with the privkeys down in `run_test`?

I don't see immediatelly how this covers the 20-of-20 case instead of just 3? Am I missing something?